### PR TITLE
Improve validation integration with the file viewer

### PR DESCRIPTION
**Pull request by kmaglione**
This is on top of #641, since it relies on those changes.

### CODE REVIEW

**Comment by kmaglione** [comment]
@wagnerand Mind reviewing the front-end portions of this?

File: apps/files/templates/files/viewer.html [hunk 1]
```diff
@@ -130,6 +134,24 @@
...
+              <input type="checkbox" id="signing-hide-unnecessary" checked="true">
+              {{ _('Hide messages not required for automated signing') }}
+            </label>
+          </div>
+          <div id="signing-hide-ignored-container"
+                {% if not (validation_data and validation_data.signing_ignored_summary) %}class="js-hidden"{% endif
+            %}>
+            <label>
+              <input type="checkbox" id="signing-hide-ignored" checked="true">
+              {{ _('Hide messages present in previous version and ignored') }}
```
Comment (wagnerand) [inline apps/files/templates/files/viewer.html]: That sounds a bit confusing. Would "Hide previously ignored messages' be still accurate enough?

Comment (kmaglione) [inline apps/files/templates/files/viewer.html]: That does sound a bit better, yeah. But this text is copied from the main validation viewer. I'm hoping to get some input from UX before I change it.

**Comment by wagnerand** [comment]
I reviewed the html, js and css files and they look good to me. r+
Thanks, Kris.

@magopian would you mind reviewing the other parts? I can understand mostly what they do but I can't provide feedback on whether it's good code. Thanks!

**Comment by magopian** [comment]
That was a big one.

r+wc

**Comment by kmaglione** [comment]
Think that's everything.

**Comment by kmaglione** [comment]
5b7abf4b5c829107

**Pull request merged**